platform/api/collection.go

```diff
@@ -61,6 +61,9 @@ func JSONGetCollection[T any](path string, out *CollectionResult[T], options *Op
 			out.Items = append(out.Items, page.Items...)
 		}
 
+		pageItemsCount = len(page.Items)
```

Comment:
@pavel-georgiev  can you please try to rebase on the latest main (which has this change applied already... I thought this went in v0.66 or so)